hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/preloader/NearCachePreloader.java

```diff
@@ -224,7 +224,7 @@ private int loadKeySet(BufferingInputStream bis, DataStructureAdapter<Object, ?>
                 break;
             }
             Data key = new HeapData(payload);
-            builder.add(serializeKeys ? key : serializationService.toObject(key));
+            builder.add(serializationService.toObject(key));
```

Comment:
Too bad the keys are about to be serialised again soon. If only we had some lazy serialising/deserialising wrapper like `CachedQueryEntry` to avoid serialisations and deserialisations all through the call chain